spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/Spark3BinPackStrategy.java

```diff
@@ -67,12 +68,18 @@ public Table table() {
           .option(SparkReadOptions.FILE_OPEN_COST, "0")
           .load(table.name());
 
+      boolean requiresRepartition = !filesToRewrite.get(0).spec().equals(table.spec());
```

Comment:
Wondering, is it enough to just check if the first one is not the same as current partition spec, and not the rest?

Reply:
Yes, we just check the first file because we are guaranteed by the rewriteAction code under the hood to only give us groups of files where every file has the same spec and partition value. They are just not guaranteed to be the same as or compatible with the current spec.

Reply:
https://github.com/apache/iceberg/blob/c6e68d832113266994c0eb06d342e032fc9cbca3/core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java#L258

Reply:
I should also note that this will end up grouping all files without compatible partition info into the same "partition" which will get sub divided randomly based on max file group size for rewrite operations.

Reply:
Sounds good, do you think its worthwhile to add a comment, if not already?

Reply:
After thinking about this for a while I think this is only true for V1 tables and we probably have a bug in V2 tables in our group by partition code in base rewrite action. Consider

```
So for example say I have a V1 table, Columns A, B
I first partition on a
spec1 = Identity<A>
then i drop that and partition on B
spec2 = Void<A>, Identity<B>

In V2 it works like this
spec1 = Identity<A>
spec2 = Identity<B>

In this case while the specs have the same shape, the values do not mean the same thing so our grouping is going to be incorrect.

Reply:
I personally think it would be, as it was clear to me from jut looking at this class that `filesToRewrite` were already grouped that way. But I'd leave that up to you due to concerns fo comment rot etc.